File: Examples/DSGE/SNM-TCN/Analyze.jl
# This does estimation via simulated annealing, only want point estimates.

using Pkg, CSV, DataFrames, Statistics, PrettyTables, Term
include("Setup.jl")

cd(@__DIR__)
Pkg.activate(".")

#function main()
infile = "PointEstimationResults.txt"

ind = [2,5]
θtrue = TrueParameters()[ind]
data = Matrix(CSV.read(infile, DataFrame, header=false))

θnn = data[:,ind]
θsa = data[:, ind .+ 7]

err_nn = (θnn .- θtrue')
err_sa = (θsa .- θtrue')

b_nn =  mean(err_nn, dims=1)'
b_sa = mean(err_sa, dims=1)'
r = size(data,1)
println("\n____________________________________")
println(@green "Results based on $r replications")
println("Bias")
names = ["γ", "ρ_η"]   
pretty_table([names b_nn b_sa], header=["param.", "θnn", "θsa"])

rmse_nn =  sqrt.(mean(err_nn .^2, dims=1)')
rmse_sa = sqrt.(mean(err_sa .^2, dims=1)')
println("RMSE")
pretty_table([names rmse_nn rmse_sa], header=["param", "θnn", "θsa"])



#end
#b = main()

File: Examples/SNM/FA/analyze.m
# true parameters

load fa.out;
data = fa;
theta1 = data(:,3:14);
printf("Rows: %d\n", rows(data));
dstats(theta1);
theta1 = theta1(:,5:8) - theta1(:,1:4); 
bias = mean(theta1);
rmse = sqrt(mean(theta1 .^2));
printf("BIAS\n");
disp(bias);
printf("RMSE\n");
disp(rmse);

